### ISSUE TITLE
Allow safely probing for surrounding _InheritedNestedScrollView

### ISSUE THREAD

**Issue by clragon** [flutter/flutter#149363]
### Use case

I find myself often using `NestedScrollView.sliverOverlapAbsorberHandleFor` lately, as I am adding `SliverAppBar`s to my App.
An immediate problem presents itself:

I always have to manually inject a `SliverOverlapAbsorber` and a `SliverOverlapInjector` whenever I am in a `NestedScrollView`,
because there is no convenient way to automatically detect a surrounding `NestedScrollView` and inject a `SliverOverlapAbsorber` with a `NestedScrollView.sliverOverlapAbsorberHandleFor`.

This is because, were I to access `NestedScrollView.sliverOverlapAbsorberHandleFor`  in a context where no `NestedScrollView` is present, an AssertionError is thrown. 

### Proposal

It would be really nice and convenient if there was an inbuilt way for me to check for an enclosing `NestedScrollView`.
Preferrably, this check would rely on looking for a `_InheritedNestedScrollView` in the tree.

I can of course right now do a very slow Widget tree search for a `NestedScrollView` manually, and then based on that decide to access `NestedScrollView.sliverOverlapAbsorberHandleFor`.

Ideally, there would be something like a `NestedScrollViewSliverOverlapAbsorber` and a accompaning `NestedScrollViewSliverOverlapInjector`, which both conditionally inject the required sliver widgets when used inside a `NestedScrollView`, but I would also be okay writing those myself.

Basically I am trying to reduce the boilerplate of getting a `SliverAppBar` setup in a `NestedScrollView`, without having to crawl through the widget tree and ruining my render performance.

**Comment by Piinks** [comment]
> were I to access NestedScrollView.sliverOverlapAbsorberHandleFor in a context where no NestedScrollView is present, an AssertionError is thrown.

Maybe what we need here is a NestedScrollView.sliverOverlapAbsorberHandleMaybeFor method that has a nullable return type? That way you would know whether or not to insert the overlap injector.

**Comment by clragon** [comment]
Certainly, that would be a solution.

Unfortunately, neither `SliverOverlapAbsorber` nor `SliverOverlapInjector` take a nullable callback, so I will have to conditionally insert or remove them.
